apps/website/src/data/network.ts

```diff
@@ -33,442 +32,436 @@ type NetworkItemCamera = NetworkItemCore & {
 
 type NetworkItemAccessPoint = NetworkItemCore & {
   type: "accessPoint";
-  links?: NetworkItem[];
+  links?: NestedNetworkItem[]; // Some Ubiquiti APs have a second ethernet port
+};
+
+type NetworkItemServer = NetworkItemCore & {
+  type: "server";
+  links?: NestedNetworkItem[]; // Allows the Cloud OBS to be at the root of the network
 };
 
-export type NetworkItem =
+export type RootNetworkItem =
   | NetworkItemSwitch
   | NetworkItemCamera
-  | NetworkItemAccessPoint;
+  | NetworkItemAccessPoint
+  | NetworkItemServer;
+export type NestedNetworkItem = RootNetworkItem & {
+  connection: NetworkConnection;
+};
+export type NetworkItem = NestedNetworkItem | RootNetworkItem;
+
+export const isNetworkItem = (item: unknown): item is NetworkItem => {
+  return (
+    !!item &&
+    typeof item === "object" &&
+    "type" in item &&
+    typeof item.type === "string" &&
+    "name" in item &&
+    typeof item.name === "string" &&
+    "model" in item &&
+    typeof item.model === "string"
+  );
+};
+
+export const isRootNetworkItem = (item: unknown): item is RootNetworkItem => {
+  return (
+    isNetworkItem(item) &&
+    (!("connection" in item) || typeof item.connection === "undefined")
+  );
+};
+
+export const isNestedNetworkItem = (
+  item: unknown,
+): item is NestedNetworkItem => {
+  return (
+    isNetworkItem(item) &&
+    "connection" in item &&
+    typeof item.connection === "object" &&
+    typeof item.connection.type === "string"
```

Comment:
`typeof item.connection.type === "string"` should always be true

Reply:
I'm not sure I follow? The value is unknown, this is a type guard, so we need to check?

Reply:
Checking `"connection" in item` should be sufficient as a check though. I dont think you need to test the type of the type property. At least my typescript is happy to narrow the type without it.
```typescript
if (!("connection" in targetNode.data.item))
  throw new Error("Invalid target");
const targetConnection = targetNode.data.item.connection;
```
is fine. You could simplify the guard to `export const isNestedNetworkItem = (item: NetworkItem): item is NestedNetworkItem => "connection" in item;` even, since the `targetNode.data.item` is already known to be a `NetworkItem`.

But its fine with the extra checks too.

Reply:
Ahh okay yeah, so really the `isNetworkItem` check should have a "if connection does exist, check type is string within it". I'm just technically doing the check at the wrong point.

Reply:
I mean the first check narrows the type down to a nested node and that guarantees the type to be a string already since the type NetworkConnection always has a type (with type string). So unless you add another type of node also with a connection property, checking for the string should not be necessary.